src/cockpit/beiboot.py

```diff
@@ -332,8 +331,18 @@ async def run(args) -> None:
         bridge.write_control(**message)
         bridge.ssh_peer.thaw_endpoint()
     except ferny.InteractionError as exc:
-        sys.exit(str(exc))
+        logger.debug("ferny.InteractionError: %s", exc)
+        msg = str(exc)
+        if 'Permission denied' in msg:
+            problem = 'authentication-failed'
+        elif 'resolve host' in msg:
```

Comment:
There's already code inside of ferny to do this and we should definitely use it.

Reply:
The `Session` API [does this](https://github.com/allisonkarlitskaya/ferny/blob/main/src/ferny/session.py#L160), [get_exception_for_ssh_stderr](https://github.com/allisonkarlitskaya/ferny/blob/main/src/ferny/errors.py#L102)

Reply:
Converted to the ferny.errors API.